controllers/bucket_controller.go

```diff
@@ -75,10 +76,32 @@ func (b BucketReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	if err != nil {
 		return result, err
 	}
+	if bucket.Annotations[oadpCloudStorageDeleteAnnotation] == "true" && bucket.DeletionTimestamp != nil {
```

Comment:
What happens if Deletion Timestamp is not nil AND no annotation?

what happens if the annotation is True, or t or something like that? do we think that we should parse it for a boolean?

Reply:
Deletion Timestamp is not nil AND no annotation
- Delete() is not called, object still contains finalizer so it will live in the cluster.
- User have two choices, remove the finalizer to remove Custom Resource without deleting cloud provider bucket  OR add annotation so that we processes the bucket deletion request with cloud provider, if successful, finalizer will be removed for user.

Reply:
@shawn-hurley do you mean
```suggestion
	if bucket.Annotations[oadpCloudStorageDeleteAnnotation] == true && bucket.DeletionTimestamp != nil {
```
or
```suggestion
	if strings.ToLower(bucket.Annotations[oadpCloudStorageDeleteAnnotation]) == "true" && bucket.DeletionTimestamp != nil {
```

Reply:
> I think I was thinking of using strconv.ParseBool helper
Sounds like a great idea.

> Wouldn't the rest of the reconcile be called? do we want that behavior if the deletion timestamp is not nil but we are not going to clean it up?
I see what you mean. How does this sound?
```suggestion
	if bucket.DeletionTimestamp != nil {
	     if !strconv.ParseBool(bucket.Annotations[oadpCloudStorageDeleteAnnotation]) {
	         return ctrl.Result{Requeue: false}, nil
	     }
	
```